That's a legit error... the precommit hook is linted apparently.

```/home/travis/build/Automattic/jetpack/bin/pre-commit-hook.js

  116:1   error  Missing JSDoc for parameter 'phpFiles'             valid-jsdoc

  119:27  error  'phpFiles' is already declared in the upper scope  no-shadow

  141:1   error  Missing JSDoc for parameter 'exitCode'             valid-jsdoc

  144:16  error  'exitCode' is already declared in the upper scope  no-shadow```

--- a/bin/phpcs-changed.sh
+++ b/bin/phpcs-changed.sh
@@ -1,0 +1,36 @@
+#!/bin/bash
+
+# This is meant to run only if PHP files have been staged. It'll fatal as of now if phpcs-changed can't find php files in the diff.
+function phpcschanged {
+  # We need three files for phpcs-changed.
+  # 1. A file of the combined diff.
+  # 2. A file of the phpcs results for the original files.
+  # 3. A file of the phpcs results for the revised files.
+
+  # Make a folder for the files.
+  mkdir tmp-phpcschanged
+  mkdir tmp-phpcschanged/original
+
+  # File 1
+  git diff --cached > tmp-phpcschanged/diff.diff
+
+  # File 2. This one is a bit fun to make.
+
+  for file in $( git diff --name-only --cached ); do
+	  mkdir -p tmp-phpcschanged/original/$(dirname ${file} ) > /dev/null 2>&1
+	  git cat-file -p master:${file} > tmp-phpcschanged/original/${file} > /dev/null 2>&1
+  done
+
+  vendor/bin/phpcs --report=json tmp-phpcschanged/original > tmp-phpcschanged/phpcs-orig.json
+
+  # File 3
+  vendor/bin/phpcs --report=json $(git diff --name-only --cached) > tmp-phpcschanged/phpcs-new.json
+
+  # Run the actual change tool!
+  vendor/bin/phpcs-changed --diff tmp-phpcschanged/diff.diff --phpcs-orig tmp-phpcschanged/phpcs-orig.json --phpcs-new tmp-phpcschanged/phpcs-new.json
+
+  # Clean up! Clean up! Everybody, eveywhere! Clean up! Clean up! Everybody does their share.
+ rm -rf tmp-phpcschanged
+}
+
+phpcschanged

--- a/bin/pre-commit-hook.js
+++ b/bin/pre-commit-hook.js
@@ -58,7 +58,8 @@ function checkFailed() {
 		chalk.red( 'COMMIT ABORTED:' ),
 		'The linter reported some problems. ' +
 			'If you are aware of them and it is OK, ' +
-			'repeat the commit command with --no-verify to avoid this check.'
+			'repeat the commit command with --no-verify to avoid this check. ' +
+			"But please don't. Code is poetry."
 	);
 	exitCode = 1;
 }
@@ -188,11 +189,18 @@ let phpChangedResult;
 if ( phpFiles.length > 0 ) {
 	phpChangedResult = spawnSync( 'composer', [ 'php:changed' ], {
 		shell: true,
-		stdio: 'inherit',
+		stdio: 'pipe',
+		encoding: 'utf-8',
 	} );
 }
 
 if ( phpChangedResult && phpChangedResult.stdout ) {
+	let phpChangedResultText;
+	phpChangedResultText = phpChangedResult.stdout.toString().split( '\n' );
+	phpChangedResultText.shift();
+	console.log(
+		JSON.stringify( JSON.parse( phpChangedResultText.toString().slice( 0, -1 ) ), null, 2 )
+	);
 	checkFailed();
 }
 

--- a/bin/pre-commit-hook.js
+++ b/bin/pre-commit-hook.js
@@ -184,6 +184,18 @@ if ( phpcsResult && phpcsResult.status ) {
 	exitCode = 1;
 }
 
+let phpChangedResult;
+if ( phpFiles.length > 0 ) {
+	phpChangedResult = spawnSync( 'composer', [ 'php:changed' ], {
+		shell: true,
+		stdio: 'inherit',
+	} );
+}
+
+if ( phpChangedResult && phpChangedResult.status ) {
+	checkFailed();
+}
+
 capturePreCommitDate();
 
 process.exit( exitCode );

--- a/bin/pre-commit-hook.js
+++ b/bin/pre-commit-hook.js
@@ -192,7 +192,7 @@ if ( phpFiles.length > 0 ) {
 	} );
 }
 
-if ( phpChangedResult && phpChangedResult.status ) {
+if ( phpChangedResult && phpChangedResult.stdout ) {
 	checkFailed();
 }
 

--- a/composer.json
+++ b/composer.json
@@ -1,7 +1,7 @@
 {
 	"name": "automattic/jetpack",
 	"description": "Jetpack supercharges your self‑hosted WordPress site with the awesome cloud power of WordPress.com",
-	"homepage": "https://jetpack.com/",
+	"homepage": "https://jetpack.com/....",
 	"type": "wordpress-plugin",
 	"license": "GPL-2.0-or-later",
 	"support": {
@@ -29,13 +29,15 @@
 		"dealerdirect/phpcodesniffer-composer-installer": "0.5.0",
 		"wp-coding-standards/wpcs": "2.1.1",
 		"sirbrillig/phpcs-variable-analysis": "2.7.0",
-		"phpcompatibility/phpcompatibility-wp": "2.1.0"
+		"phpcompatibility/phpcompatibility-wp": "2.1.0",
+		"sirbrillig/phpcs-changed": "^1.0"
 	},
 	"scripts": {
 		"php:compatibility": "composer install && vendor/bin/phpcs -p -s --runtime-set testVersion '5.6-' --standard=PHPCompatibilityWP --ignore=docker,tools,tests,node_modules,vendor --extensions=php",
 		"php:lint": "composer install && vendor/bin/phpcs -p -s",
 		"php:autofix": "composer install && vendor/bin/phpcbf",
-		"php:lint:errors": "composer install && vendor/bin/phpcs -p -s --runtime-set ignore_warnings_on_exit 1"
+		"php:lint:errors": "composer install && vendor/bin/phpcs -p -s --runtime-set ignore_warnings_on_exit 1",
+		"php:changed": "composer install && bin/phpcs-changed.sh"
 	},
 	"repositories": [
 		{

